### PR TITLE
[#33] Diameter & Radius Confusion

### LINKED ISSUE #33: Radius parameter is actually a diameter
CircularPercentIndicator has a property `final double radius;` but in implementation, it is used as a diameter:
```
Container(
        height: widget.radius + widget.lineWidth,
        width: widget.radius,
```
Here we get the real radius by dividing by two
```
CirclePainter(
              radius: (widget.radius / 2) - widget.lineWidth / 2,
```
Please read more info at link https://www.mathlearnit.com/circle-circumference.html
circle-circumference-radius-diameter

### PR DESCRIPTION
**Pull request by nvshah**
Fix #33 :- facilitate true radius for CircularPercentIndicator

### CODE REVIEW

**Comment by diegoveloper** [comment]
Can you run the samples inside this package and add some screenshots to see how those looks like ?

**Comment by nvshah** [comment]
@diegoveloper 

Please find the snaps as attached below : 

Circular_Snap1

Circular_Snap2
> NOTE :- In second Image, as radius goes beyond the 400, so it shows overflow !

Circular_Snap3

**Comment by diegoveloper** [comment]
> @diegoveloper
> 
> Please find the snaps as attached below :
> 
> Circular_Snap1
> 
> Circular_Snap2
> 
> > NOTE :- In second Image, as radius goes beyond the 400, so it shows overflow !
> 
> Circular_Snap3

Yes, as you can see it's breaking the current behavior, it will happen the same with all of the current users/devs that are using this package.

This is a breaking change, can you also update the current samples? and use the half of the current "radius" (the sample should looks the same as before this change ).

**Comment by nvshah** [comment]
@diegoveloper 
Yes when radius is taken half, then it resembles same as earlier
Should I attach snaps for same ??

**Comment by diegoveloper** [comment]
> 

yes please :)

**Comment by nipun-solutelabs** [comment]
@diegoveloper 

Please find images with radius = half of what above taken

**Comment by diegoveloper** [comment]
ok, it looks good, don't forget to push your changes.

**Comment by nvshah** [comment]
Done (pushed sample dart file changes for half-radius as well)

File: lib/circular_percent_indicator.dart [hunk 1]
```diff
@@ -253,7 +255,7 @@ class _CircularPercentIndicatorState extends State<CircularPercentIndicator>
                 backgroundColor: widget.backgroundColor,
                 startAngle: widget.startAngle,
                 circularStrokeCap: widget.circularStrokeCap,
-                radius: (widget.radius / 2) - widget.lineWidth / 2,
+                radius: widget.radius - widget.lineWidth / 2,
```
Comment (diegoveloper) [inline lib/circular_percent_indicator.dart:258]: avoid using widget.radius if we already have a `_diameter` variable

File: lib/circular_percent_indicator.dart [hunk 2]
```diff
@@ -286,7 +288,7 @@ class _CircularPercentIndicatorState extends State<CircularPercentIndicator>
                         (widget.circularStrokeCap != CircularStrokeCap.butt)
                             ? widget.lineWidth / 2
                             : 0,
-                        (-widget.radius / 2 + widget.lineWidth / 2),
+                        (-widget.radius + widget.lineWidth / 2),
```
Comment (diegoveloper) [inline lib/circular_percent_indicator.dart:291]: same here, avoid using widget.radius

File: lib/circular_percent_indicator.dart (hunk 1 above)
Comment (nvshah) [reply lib/circular_percent_indicator.dart:258]: @diegoveloper, 
I'll make necessary changes, but I don't understand motif as we already have radius then why do we again compute radius from (_diameter / 2)

Comment (diegoveloper) [reply lib/circular_percent_indicator.dart:258]: then we don't need a `_diameter` variable  after all, right?

Comment (nvshah) [reply lib/circular_percent_indicator.dart:258]: _diameter variable is created for Container ( as Container takes height & width ),
so given radius -> we compute diameter (as the only place where diameter is used is in Container size)
elsewhere -> radius itself is used

Comment (nvshah) [reply lib/circular_percent_indicator.dart:258]: _diameter can omitted if 
we replace it with `_radius * 2`

Comment (diegoveloper) [reply lib/circular_percent_indicator.dart:258]: you can use widget.radius*2 for the container

Comment (diegoveloper) [reply lib/circular_percent_indicator.dart:258]: but ok, I understand, that's fine then

Comment (nvshah) [reply lib/circular_percent_indicator.dart:258]: But I thought if its used at 2 places so if needed in Future case also so created new variable for ease !

**Review by diegoveloper — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**